File: worthly-mobile/lib/theme/categories.dart
import 'package:worthly_mobile/api/models.dart';

List<Category> parentCategories(List<Category> categories) {
  return categories
      .where((item) => item.system && item.code != null && item.code != 'uncategorized' && !(item.code?.startsWith('transfer.') ?? false))
      .toList();
}

String defaultParentId(List<Category> categories) {
  final parents = parentCategories(categories);
  return parents.where((item) => item.code == 'expense.other').firstOrNull?.id ?? parents.firstOrNull?.id ?? '';
}

List<Category> customCategories(List<Category> categories) {
  return categories.where((item) => !item.system).toList();
}

File: worthly-web/src/lib/categories.ts
import type { Category } from "@/lib/types";

export function parentCategories(categories: Category[]): Category[] {
  return categories.filter(
    (item) => item.system && item.code && item.code !== "uncategorized" && !item.code.startsWith("transfer."),
  );
}

export function defaultParentId(categories: Category[]): string {
  const parents = parentCategories(categories);
  return parents.find((item) => item.code === "expense.other")?.id ?? parents[0]?.id ?? "";
}

export function customCategories(categories: Category[]): Category[] {
  return categories.filter((item) => !item.system);
}
